### PR TITLE
Payload to be passed as interface{} into transaction instead of string

### PR DESCRIPTION
**Pull request by Dmdv**
### Changes
-

### Fixes
-

### Tests
Tasks to complete before merging PR:
- [ ]  Ensure system tests are passing. If not [Run them manually](https://github.com/0chain/gosdk/actions/workflows/system_tests.yml) to check for any regressions :clipboard:
- [ ]  Do any new system tests need added to test this change? do any existing system tests need updated? If so create a PR at [0chain/system_test](https://github.com/0chain/system_test)
- [ ]  Merge your system tests PR to master AFTER merging this PR

### Associated PRs (Link as appropriate):
- blobber:
- 0chain:
- system_test:
- zboxcli:
- zwalletcli:
- Other:

https://github.com/0chain/gosdk/issues/453
https://github.com/0chain/0chain/issues/1238
https://github.com/0chain/token_bridge_authserver/pull/13/

### CODE REVIEW

File: zcnbridge/transaction/functions.go [hunk 1]
```diff
@@ -1,36 +1,29 @@
...
 	}
 
-	_, err = t.ExecuteSmartContract(
-		ctx,
-		zcncore.ZCNSCSmartContractAddress,
-		transaction.ZCNSC_ADD_AUTHORIZER,
-		string(buffer),
-		0,
-	)
+	err = t.callBack.waitCompleteCall(context.Background())
```
Comment (dabasov) [inline zcnbridge/transaction/functions.go]: add timeout here

File: zcnbridge/transaction/functions.go [hunk 2]
```diff
@@ -1,36 +1,29 @@
...
 import (
 	"context"
-	"encoding/json"
 
-	"github.com/0chain/gosdk/core/transaction"
 	"github.com/0chain/gosdk/zcncore"
 )
 
-func (t *Transaction) AddAuthorizer(ctx context.Context, input *zcncore.AddAuthorizerPayload) error {
+func AddAuthorizer(input *zcncore.AddAuthorizerPayload) (*Transaction, error) {
```
Comment (dabasov) [inline zcnbridge/transaction/functions.go]: add context

File: zcnbridge/transaction/txn.go [hunk 3]
```diff
@@ -70,7 +70,7 @@ func NewTransactionEntity() (*Transaction, error) {
 // ExecuteSmartContract executes function of smart contract with provided address.
 //
 // Returns hash of executed transaction.
-func (t *Transaction) ExecuteSmartContract(ctx context.Context, address, funcName, input string, val int64) (string, error) {
+func (t *Transaction) ExecuteSmartContract(ctx context.Context, address, funcName string, input interface{}, val int64) (string, error) {
```
Comment (dabasov) [inline zcnbridge/transaction/txn.go:73]: why do we need it and does not use appropriate funcs from gosdk, as for ZCNSCAddAuthorizer?

File: zcnbridge/transaction/functions.go (hunk 1 above)
Comment (Dmdv) [reply zcnbridge/transaction/functions.go]: Added the context but the timeout is on the side of a client

File: zcnbridge/transaction/functions.go (hunk 2 above)
Comment (Dmdv) [reply zcnbridge/transaction/functions.go]: added context

File: zcnbridge/transaction/txn.go (hunk 3 above)
Comment (Dmdv) [reply zcnbridge/transaction/txn.go:73]: It actually does

File: core/conf/vars.go [hunk 4]
```diff
@@ -48,10 +49,22 @@ func InitChainNetwork(n *Network) {
 
 	if network == nil {
 		network = n
+		normalizeURLs(network)
```
Comment (dabasov) [inline core/conf/vars.go]: can remove it

File: zcnbridge/transaction/functions.go [hunk 5]
```diff
@@ -1,36 +1,29 @@
...
 
-	_, err = t.ExecuteSmartContract(
-		ctx,
-		zcncore.ZCNSCSmartContractAddress,
-		transaction.ZCNSC_ADD_AUTHORIZER,
-		string(buffer),
-		0,
-	)
+	err = t.callBack.waitCompleteCall(ctx)
+	t.Hash = t.scheme.Hash()
```
Comment (dabasov) [inline zcnbridge/transaction/functions.go:23]: is it really needed?

File: core/conf/vars.go (hunk 4 above)
Comment (Dmdv) [reply core/conf/vars.go]: This is actually a fix

File: zcnbridge/transaction/functions.go (hunk 5 above)
Comment (Dmdv) [reply zcnbridge/transaction/functions.go:23]: Yes, required for further transaction verification

File: zcnbridge/transaction/txn.go [hunk 6]
```diff
@@ -84,22 +86,36 @@ func (t *Transaction) ExecuteSmartContract(ctx context.Context, address, funcNam
...
 	}
 
 	return t.Hash, nil
 }
 
-// Verify checks including of transaction in the blockchain.
-func (t *Transaction) Verify(ctx context.Context) error {
-	const errCode = "transaction_verify"
+func (t *Transaction) GetTransactionHash() string {
+	var txnout map[string]json.RawMessage
```
Comment (peterlimg) [inline zcnbridge/transaction/txn.go]: The transaction hash should exist once the transaction is created and signed, we don't need to parse the transaction result to get it.

**Review by peterlimg — DISMISSED** [review]

File: zcnbridge/transaction/txn.go (hunk 6 above)
Comment (Dmdv) [reply zcnbridge/transaction/txn.go]: I've removed it

File: zcnbridge/transaction/functions.go (hunk 1 above)
Comment (dabasov) [reply zcnbridge/transaction/functions.go]: we should avoid using background context, add default timeout context

File: zcnbridge/transaction/functions.go (hunk 5 above)
Comment (dabasov) [reply zcnbridge/transaction/functions.go:23]: but Transaction has 2 different hash fields, one in t.scheme and another in t. 
Why we didn't embed t.scheme in transaction, so we could remove hash field duplication and use something like this?
```go
func (t *Transaction) Hash() string {
    return t.scheme.Hash()
 
}
```

**Review by peterlimg — APPROVED** [review]
lgtm

**Pull request merged**